main.py

```diff
@@ -470,37 +503,16 @@ def main(cl_arguments):
 
     if args.do_target_task_training:
         # Train on target tasks
-        task_names_to_avoid_loading = setup_target_task_training(args, target_tasks, model, strict)
-        if args.transfer_paradigm == "frozen":
-            # might be empty if elmo = 0. scalar_mix_0 should always be
-            # pretrain scalars
-            elmo_scalars = [
-                (n, p)
-                for n, p in model.named_parameters()
-                if "scalar_mix" in n and "scalar_mix_0" not in n
-            ]
-            # Fails when sep_embs_for_skip is 0 and elmo_scalars has nonzero
-            # length.
-            assert_for_log(
-                not elmo_scalars or args.sep_embs_for_skip,
-                "Error: ELMo scalars loaded and will be updated in do_target_task_training but "
-                "they should not be updated! Check sep_embs_for_skip flag or make an issue.",
-            )
+        pre_target_train_path = setup_target_task_training(args, target_tasks, model, strict)
+
         for task in target_tasks:
             # Skip diagnostic tasks b/c they should not be trained on
             if isinstance(task, GLUEDiagnosticTask):
                 continue
 
-            if args.transfer_paradigm == "finetune":
-                # Train both the task specific models as well as sentence
-                # encoder.
-                to_train = [(n, p) for n, p in model.named_parameters() if p.requires_grad]
-            else:  # args.transfer_paradigm == "frozen":
-                # Only train task-specific module
-                pred_module = getattr(model, "%s_mdl" % task.name)
-                to_train = [(n, p) for n, p in pred_module.named_parameters() if p.requires_grad]
-                to_train += elmo_scalars
-
+            pre_target_train_model_params = load_model_for_target_train_run(
```

Comment:
This suggests that it could include all the params. How about 'params_to_train'?